### PR TITLE
adding compaction_written_bytes/sec stats to scorch

### PR DESCRIPTION
**Pull request by sreekanth-cb**
I wasn't sure whether we need the written bytes to be tracked at the CountHashWriter write method level or not.

### CODE REVIEW

File: index/scorch/stats.go [hunk 1]
```diff
@@ -100,6 +100,8 @@ type Stats struct {
 	TotMemMergeZapBeg   uint64
 	TotMemMergeZapEnd   uint64
 	TotMemMergeSegments uint64
+
+	TotCompactionWrittenBytes uint64
```
Comment (steveyen) [inline index/scorch/stats.go]: naming thin -- i think in scorch we don't usually use the word "compaction", even though that's a legit way to think of what the file merger is doing.

How about "TotFileMergeWrittenBytes", and move this stat up a few lines into the file merge related clumping of stats?  (but keep that backwards compat naming alias elsewhere that's "total_compaction_written_bytes"?

**Comment by steveyen** [comment]
Hi @sreekanth-cb -- another thing (capturing something @mschoch mentioned)... not sure we need an additional StatsReporter interface for this?  That is, an interface or level-of-indirection might be warranted if there's different kinds of stats receivers, but in this case there's just that one stats struct?

File: index/scorch/merge.go [hunk 2]
```diff
@@ -183,16 +183,17 @@ func (s *Scorch) planMergeAtSnapshot(ourSnapshot *IndexSnapshot,
 			fileMergeZapStartTime := time.Now()
 
 			atomic.AddUint64(&s.stats.TotFileMergeZapBeg, 1)
-			newDocNums, err := zap.Merge(segmentsToMerge, docsToDrop, path, 1024)
+			newDocNums, nBytes, err := zap.Merge(segmentsToMerge, docsToDrop, path, 1024)
 			atomic.AddUint64(&s.stats.TotFileMergeZapEnd, 1)
-
+      atomic.AddUint64(&s.stats.TotFileMergeWrittenBytes, nBytes)
```
Comment (mschoch) [inline index/scorch/merge.go]: wondering why the indentation doesn't match here, do we need to run go fmt?

File: index/scorch/merge.go [hunk 3]
```diff
@@ -183,16 +183,17 @@ func (s *Scorch) planMergeAtSnapshot(ourSnapshot *IndexSnapshot,
...
+      
 			fileMergeZapTime := uint64(time.Since(fileMergeZapStartTime))
 			atomic.AddUint64(&s.stats.TotFileMergeZapTime, fileMergeZapTime)
 			if atomic.LoadUint64(&s.stats.MaxFileMergeZapTime) < fileMergeZapTime {
 				atomic.StoreUint64(&s.stats.MaxFileMergeZapTime, fileMergeZapTime)
 			}
-
-			if err != nil {
+
```
Comment (mschoch) [inline index/scorch/merge.go]: seems like indentation changed here as well

**Pull request merged**